packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -3,272 +3,122 @@ import { unstable_useControlled as useControlled } from '@mui/utils';
-  /**
-   * The selected value.
-   * Used when the component is controlled.
-   */
-  value?: TValue;
-  /**
-   * The default value.
-   * Used when the component is not controlled.
-   */
-  defaultValue?: TValue;
-  /**
-   * Callback fired when the value changes.
-   * @template TValue The value type. Will be either the same type as `value` or `null`. Can be in `[start, end]` format in case of range value.
-   * @template TError The validation error type. Will be either `string` or a `null`. Can be in `[start, end]` format in case of range value.
-   * @param {TValue} value The new value.
-   * @param {FieldChangeHandlerContext<TError>} context The context containing the validation result of the current value.
-   */
-  onChange?: PickerChangeHandler<TValue, TError>;
-  /**
-   * Callback fired when the value is accepted.
-   * @template TValue The value type. Will be either the same type as `value` or `null`. Can be in `[start, end]` format in case of range value.
-   * @param {TValue} value The value that was just accepted.
-   */
-  onAccept?: (value: TValue) => void;
-  /**
-   * Callback fired when the error associated to the current value changes.
-   * If the error has a non-null value, then the `TextField` will be rendered in `error` state.
-   *
-   * @template TValue The value type. Will be either the same type as `value` or `null`. Can be in `[start, end]` format in case of range value.
-   * @template TError The validation error type. Will be either `string` or a `null`. Can be in `[start, end]` format in case of range value.
-   * @param {TError} error The new error describing why the current value is not valid.
-   * @param {TValue} value The value associated to the error.
-   */
-  onError?: (error: TError, value: TValue) => void;
-}
+const shouldPublishValue = <TValue, TError>(
+  params: PickerValueUpdaterParams<TValue, TError>,
+): boolean => {
+  const { action, hasChanged, dateState, isControlled } = params;
 
-/**
- * Props used to handle the value of non-static pickers.
- */
-export interface UsePickerValueNonStaticProps<TValue, TSection extends FieldSection>
-  extends Pick<
-    UseFieldInternalProps<TValue, TSection, unknown>,
-    'selectedSections' | 'onSelectedSectionsChange'
-  > {
-  /**
-   * If `true`, the popover or modal will close after submitting the full date.
-   * @default `true` for desktop, `false` for mobile (based on the chosen wrapper and `desktopModeMediaQuery` prop).
-   */
-  closeOnSelect?: boolean;
-  /**
-   * Control the popup or dialog open state.
-   * @default false
-   */
-  open?: boolean;
-  /**
-   * Callback fired when the popup requests to be closed.
-   * Use in controlled mode (see `open`).
-   */
-  onClose?: () => void;
-  /**
-   * Callback fired when the popup requests to be opened.
-   * Use in controlled mode (see `open`).
-   */
-  onOpen?: () => void;
-}
+  const isCurrentValueTheDefaultValue = !isControlled && !dateState.hasBeenModifiedSinceMount;
```

Comment:
Personal note: Verify how `hasBeenModifiedSinceMount` is managed. Especially for cases with multiple views, such as year > date > hours > minutes where we could multiple time select the default value